client/web/web.go

```diff
@@ -1108,6 +1140,285 @@ func (s *Server) servePostRoutes(ctx context.Context, data postRoutesRequest) er
 	return err
 }
 
+type serveItem struct {
+	Target      serveTarget      `json:"target"`
+	Destination serveDestination `json:"destination"`
+	ShareType   string           `json:"shareType"` // "serve" or "funnel"
+
+	IsForeground bool `json:"isForeground,omitempty"` // only populated by "GET", empty for "PATCH"/"DELETE"
+	IsEdit       bool `json:"isEdit,omitempty"`       // only populated by "PATCH", true when editing an existing item, false when adding a new one
+}
+
+type serveTarget struct {
+	Type  string `json:"type"`  // "plainText" or "localHttpPort"
+	Value string `json:"value"` // Any text if type is "plainText"; port number if type is "localHttpPort"
+}
+
+type serveDestination struct {
+	Protocol string `json:"protocol"` // "https", "http", "tcp", or "tls-terminated-tcp"
+	Port     uint16 `json:"port"`     // 443 or 8443 or 10000
+	Path     string `json:"path"`     // e.g. /images/dogs; only for "https" or "http"
+}
+
+func (s *Server) serveGetServeItems(w http.ResponseWriter, r *http.Request) {
```

Comment:
serve seems to be a significant enough feature in terms of logic we have to implement, do you think it's worth pulling out into a separate `serve.go` file?